staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -243,6 +243,7 @@ func (d *dialerCreator) createDialer() utilnet.DialFunc {
 		ctx, span := tracing.Start(ctx, fmt.Sprintf("Proxy via %s protocol over %s", d.options.protocol, d.options.transport), attribute.String("address", addr))
 		defer span.End(500 * time.Millisecond)
 		start := egressmetrics.Metrics.Clock().Now()
+		egressmetrics.Metrics.ObserveDialStart(d.options.protocol, d.options.transport)
```

Comment:
is the idea that comparing this count to the latency/failure counts would reveal a hang/leak? since latencies is a histogram, can it be used to get a comparable count?

Reply:
That's right, and it was my motivation here. But it's also nice to have as a denominator (for success or failure rate, since the latency and failure metrics are mutually exclusive).

Reply:
and the latency histogram metric doesn't do summarization in a way that would prevent us from extracting the counts to compare, right?

Reply:
as long as we *can* compute the total completed+failed given the other metrics we have, this lgtm. an ack from @logicalhan on that would be helpful

Reply:
Yeah the histogram should give us total counts.